sdk/python/kfp/compiler/compiler.py

```diff
@@ -394,10 +472,22 @@ def _group_to_template(self, group, inputs, outputs, dependencies):
             })
           else:
             # The value comes from its parent.
-            arguments.append({
-              'name': param_name,
-              'value': '{{inputs.parameters.%s}}' % param_name
-            })
+            # Special handling for recursive subgroup: argument name comes from the existing opsgroup
```

Comment:
Why do you need the special handling here? Can the recursive opsgroup have same param name as the recursive_ref?

Reply:
The argument name for the recursive opsgroup step in the outlayer opsgroup should consistent with the parameter name in the recursive opsgroup template, which is generated based on the input name in the recursive_ref opsgroup object.

Reply:
The reason that they differ is because the input name in the recursive_ref opsgroup template is generated based on the pipelineparam name outside of the opsgroup.

Reply:
I see. Thanks.